### PR TITLE
CLOUDP-228589: Migrate AuditLog to new Atlas SDK

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/controller/workflow/reason.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ const (
 	ProjectNetworkPeerIsNotReadyInAtlas        ConditionReason = "ProjectNetworkPeerIsNotReadyInAtlas"
 	ProjectEncryptionAtRestReady               ConditionReason = "ProjectEncryptionAtRestReady"
 	ProjectCloudIntegrationsIsNotReadyInAtlas  ConditionReason = "ProjectCloudIntegrationsIsNotReadyInAtlas"
-	ProjectAuditingReady                       ConditionReason = "ProjectAuditingReady"
+	ProjectAuditLogsNotConfigured              ConditionReason = "ProjectAuditLogsNotConfigured"
```
Comment (s-urbaniak) [inline pkg/controller/workflow/reason.go:41]: does this end up in status conditions? If so, this is effectively an API change.

**Comment by s-urbaniak** [comment]
one concern regarding status conditions, else great! 👍

File: internal/mocks/translation/audit.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package translation
```
Comment (josvazg) [inline internal/mocks/translation/audit.go:1]: In other PR (or PRs) I put thin under the fake folder.

We should be consistent, so lets ask for others votes here:
@s-urbaniak &  @roothorp WDYT?

`internal/resource/fake`
OR
`internal/mocks/translation` (all mocks in a single package it seems)

File: internal/translation/audit/audit.go [hunk 3]
```diff
@@ -24,15 +19,6 @@ type AuditLog struct {
 	auditAPI admin.AuditingApi
 }
 
-// NewAuditLogService creates an AuditLog from credentials and the atlas provider
```
Comment (josvazg) [inline internal/translation/audit/audit.go:27]: If we do this everywhere we might be able to remove the code at the `translation` parent package.

Maybe we should pin a reminder for that in a Jira. But it does not affect this PR.

File: pkg/controller/atlasproject/auditing.go [hunk 4]
```diff
@@ -3,96 +3,87 @@ package atlasproject
...
-	}
-
-	if isAuditingEmpty(project.Spec.Auditing) {
-		workflowCtx.UnsetCondition(api.AuditingReadyType)
-		return workflow.OK()
-	}
-
-	workflowCtx.SetConditionTrue(api.AuditingReadyType)
-	return workflow.OK()
+type auditController struct {
```
Comment (josvazg) [inline pkg/controller/atlasproject/auditing.go:12]: Great!

File: pkg/controller/workflow/reason.go (hunk 1 above)
Comment (helderjs) [reply pkg/controller/workflow/reason.go:41]: It's not a new status, just a more meaningful reason. Is it still a breaking change?

Comment (josvazg) [reply pkg/controller/workflow/reason.go:41]: +1

**Comment by josvazg** [comment]
LGTM, waiting on that API change issue.

About the mocks voting, we can merge and fix afterwards if needed, whatever the decision is.

File: pkg/controller/workflow/reason.go (hunk 1 above)
Comment (s-urbaniak) [reply pkg/controller/workflow/reason.go:41]: This is removing `ProjectAuditingReady` in favor of `ProjectAuditLogsNotConfigured`.
If users rely on `ProjectAuditingReady` in their GitOps pipeline, they will be broken after this change.
Hence, we cannot simply remove reasons from status conditions, they are meant to be machine-readable.
`message` though can be improved.

File: internal/mocks/translation/audit.go (hunk 2 above)
Comment (s-urbaniak) [reply internal/mocks/translation/audit.go:1]: this is a nuance, leaving the final call to you @josvazg and @helderjs

File: pkg/controller/atlasproject/auditing.go (hunk 4 above)
Comment (s-urbaniak) [reply pkg/controller/atlasproject/auditing.go:12]: indeed, this is a slick, small, and neat little state machine done here 😍

File: pkg/controller/atlasproject/auditing.go [hunk 5]
```diff
@@ -3,96 +3,87 @@ package atlasproject
...
+		return a.configure(auditInAKO, isUnset)
+	}
 
-func prepareAuditingSpec(spec *akov2.Auditing) *mongodbatlas.Auditing {
-	if isAuditingEmpty(spec) {
-		return &mongodbatlas.Auditing{
-			Enabled: pointer.MakePtr(false),
-		}
+	if a.project.Spec.Auditing == nil {
+		return a.unmanage()
```
Comment (s-urbaniak) [inline pkg/controller/atlasproject/auditing.go:33]: nit: this transition is not handled in unit tests.

**Review by s-urbaniak — CHANGES_REQUESTED** [review]
two concerns:
- missing unit test controller
- when I execute the contract test locally, I actually get a JSON_FILTER config type by default

File: pkg/controller/atlasproject/auditing.go (hunk 5 above)
Comment (helderjs) [reply pkg/controller/atlasproject/auditing.go:33]: Added

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**